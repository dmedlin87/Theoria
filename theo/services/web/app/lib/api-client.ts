import { getApiBaseUrl } from "./api";
import type { components } from "./generated/api";

function buildErrorMessage(status: number, body: string): string {
  if (body) {
    return body;
  }
  return `Request failed with status ${status}`;
}

async function handleResponse<T>(response: Response, parseJson: boolean): Promise<T> {
  if (!response.ok) {
    const body = await response.text();
    throw new Error(buildErrorMessage(response.status, body));
  }
  if (!parseJson) {
    return undefined as T;
  }
  if (response.status === 204) {
    return undefined as T;
  }
  return (await response.json()) as T;
}

export type RequestOptions = RequestInit & { parseJson?: boolean };

export class TheoApiClient {
  private readonly baseUrl: string;

  constructor(baseUrl?: string) {
    const resolved = (baseUrl ?? getApiBaseUrl()).replace(/\/$/, "");
    this.baseUrl = resolved;
  }

  private async request<T>(path: string, init?: RequestOptions): Promise<T> {
    const { parseJson = true, headers, ...rest } = init ?? {};
    const response = await fetch(`${this.baseUrl}${path}`, {
      cache: "no-store",
      headers: {
        "Content-Type": "application/json",
        ...(headers ?? {}),
      },
      ...rest,
    });
    return handleResponse<T>(response, parseJson);
  }

  fetchFeatures(): Promise<Record<string, boolean>> {
    return this.request<Record<string, boolean>>("/features/");
  }

  runVerseWorkflow(payload: {
    model: string;
    osis?: string | null;
    passage?: string | null;
    question?: string | null;
  }): Promise<import("../copilot/components/types").VerseResponse> {
    return this.request("/ai/verse", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runSermonWorkflow(payload: {
    model: string;
    topic: string;
    osis?: string | null;
  }): Promise<import("../copilot/components/types").SermonResponse> {
    return this.request("/ai/sermon-prep", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runComparativeWorkflow(payload: {
    model: string;
    osis: string;
    participants: string[];
  }): Promise<import("../copilot/components/types").ComparativeResponse> {
    return this.request("/ai/comparative", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runMultimediaWorkflow(payload: {
    model: string;
    collection?: string | null;
  }): Promise<import("../copilot/components/types").MultimediaDigestResponse> {
    return this.request("/ai/multimedia", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runDevotionalWorkflow(payload: {
    model: string;
    osis: string;
    focus: string;
  }): Promise<import("../copilot/components/types").DevotionalResponse> {
    return this.request("/ai/devotional", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runCollaborationWorkflow(payload: {
    model: string;
    thread: string;
    osis: string;
    viewpoints: string[];
  }): Promise<import("../copilot/components/types").CollaborationResponse> {
    return this.request("/ai/collaboration", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runCurationWorkflow(payload: {
    model: string;
    since?: string | null;
  }): Promise<import("../copilot/components/types").CorpusCurationReport> {
    return this.request("/ai/curation", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  runSermonExport(payload: {
    model: string;
    topic: string;
    osis?: string | null;
    format: string;
  }): Promise<import("../copilot/components/types").ExportPresetResult> {
    return this.request("/ai/sermon-prep/export?format=" + encodeURIComponent(payload.format), {
      method: "POST",
      body: JSON.stringify({
        model: payload.model,
        topic: payload.topic,
        osis: payload.osis ?? null,
      }),
    });
  }

  runTranscriptExport(payload: {
    documentId: string;
    format: string;
  }): Promise<import("../copilot/components/types").ExportPresetResult> {
    return this.request("/ai/transcript/export", {
      method: "POST",
      body: JSON.stringify({
        document_id: payload.documentId,
        format: payload.format,
      }),
    });
  }

  exportCitations(
    payload: components["schemas"]["CitationExportRequest"],
  ): Promise<import("../copilot/components/types").CitationExportResponse> {
    return this.request("/ai/citations/export", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  getDigest(): Promise<import("../admin/digests/types").TopicDigest> {
    return this.request("/ai/digest");
  }

  refreshDigest(hours: number): Promise<import("../admin/digests/types").TopicDigest> {
    return this.request(`/ai/digest?hours=${hours}`, {
      method: "POST",
    });
  }

  listWatchlists(userId: string): Promise<import("../admin/digests/types").WatchlistResponse[]> {
    return this.request(`/ai/digest/watchlists?user_id=${encodeURIComponent(userId)}`);
  }

  createWatchlist(payload: import("../admin/digests/types").CreateWatchlistPayload): Promise<
    import("../admin/digests/types").WatchlistResponse
  > {
    return this.request("/ai/digest/watchlists", {
      method: "POST",
      body: JSON.stringify(payload),
    });
  }

  updateWatchlist(
    watchlistId: string,
    payload: import("../admin/digests/types").WatchlistUpdatePayload,
  ): Promise<import("../admin/digests/types").WatchlistResponse> {
    return this.request(`/ai/digest/watchlists/${watchlistId}`, {
      method: "PATCH",
      body: JSON.stringify(payload),
    });
  }

  deleteWatchlist(watchlistId: string): Promise<void> {
    return this.request(`/ai/digest/watchlists/${watchlistId}`, {
      method: "DELETE",
      parseJson: false,
    });
  }

  runWatchlist(
    watchlistId: string,
    type: "preview" | "run",
  ): Promise<import("../admin/digests/types").WatchlistRunResponse> {
    const path =
      type === "preview"
        ? `/ai/digest/watchlists/${watchlistId}/preview`
        : `/ai/digest/watchlists/${watchlistId}/run`;
    const method = type === "preview" ? "GET" : "POST";
    return this.request(path, { method });
  }

  fetchWatchlistEvents(
    watchlistId: string,
    since?: string,
  ): Promise<import("../admin/digests/types").WatchlistRunResponse[]> {
    const query = since ? `?since=${encodeURIComponent(since)}` : "";
    return this.request(`/ai/digest/watchlists/${watchlistId}/events${query}`);
  }
}

export function createTheoApiClient(baseUrl?: string): TheoApiClient {
  return new TheoApiClient(baseUrl);
}
