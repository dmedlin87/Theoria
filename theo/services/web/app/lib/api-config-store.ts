export type ApiCredentials = {
  authorization: string | null;
  apiKey: string | null;
};

const STORAGE_KEY = "theo.api.credentials";

let cachedCredentials: ApiCredentials | null = null;

function normalizeValue(value: string | null | undefined): string | null {
  if (typeof value !== "string") {
    return null;
  }
  const trimmed = value.trim();
  return trimmed.length > 0 ? trimmed : null;
}

export function normalizeCredentials(
  credentials: Partial<ApiCredentials> | ApiCredentials | null | undefined,
): ApiCredentials {
  return {
    authorization: normalizeValue(credentials?.authorization),
    apiKey: normalizeValue(credentials?.apiKey),
  };
}

export function getCachedCredentials(): ApiCredentials | null {
  return cachedCredentials;
}

export function setCachedCredentials(credentials: ApiCredentials | null): void {
  cachedCredentials = credentials ? normalizeCredentials(credentials) : null;
}

export function readCredentialsFromStorage(): ApiCredentials | null {
  if (typeof window === "undefined") {
    return null;
  }
  try {
    const raw = window.localStorage.getItem(STORAGE_KEY);
    if (!raw) {
      return null;
    }
    const parsed = JSON.parse(raw) as Partial<ApiCredentials> | null;
    if (!parsed || typeof parsed !== "object") {
      return null;
    }
    const normalized = normalizeCredentials(parsed);
    if (!normalized.authorization && !normalized.apiKey) {
      return null;
    }
    return normalized;
  } catch {
    return null;
  }
}

export function writeCredentialsToStorage(credentials: ApiCredentials): void {
  if (typeof window === "undefined") {
    return;
  }
  const normalized = normalizeCredentials(credentials);
  if (!normalized.authorization && !normalized.apiKey) {
    window.localStorage.removeItem(STORAGE_KEY);
    return;
  }
  try {
    window.localStorage.setItem(STORAGE_KEY, JSON.stringify(normalized));
  } catch {
    // Ignore storage errors (private browsing, quota issues, etc.)
  }
}

export function clearCredentialsStorage(): void {
  if (typeof window === "undefined") {
    return;
  }
  try {
    window.localStorage.removeItem(STORAGE_KEY);
  } catch {
    // Ignore storage errors
  }
}

function pickEnvValue(names: string[]): string | null {
  for (const name of names) {
    const value = process.env[name];
    if (typeof value === "string") {
      const normalized = value.trim();
      if (normalized) {
        return normalized;
      }
    }
  }
  return null;
}

const PUBLIC_AUTHORIZATION_ENV_VARS = ["NEXT_PUBLIC_API_AUTHORIZATION"];
const SERVER_AUTHORIZATION_ENV_VARS = [
  "API_AUTHORIZATION",
  "THEO_API_AUTHORIZATION",
];
const PUBLIC_API_KEY_ENV_VARS = ["NEXT_PUBLIC_API_KEY"];
const SERVER_API_KEY_ENV_VARS = ["API_KEY", "THEO_SEARCH_API_KEY"];

export function getEnvCredentials(): ApiCredentials {
  const isBrowser = typeof window !== "undefined";
  const authorizationNames = isBrowser
    ? PUBLIC_AUTHORIZATION_ENV_VARS
    : [...PUBLIC_AUTHORIZATION_ENV_VARS, ...SERVER_AUTHORIZATION_ENV_VARS];
  const apiKeyNames = isBrowser
    ? PUBLIC_API_KEY_ENV_VARS
    : [...PUBLIC_API_KEY_ENV_VARS, ...SERVER_API_KEY_ENV_VARS];

  return {
    authorization: pickEnvValue(authorizationNames),
    apiKey: pickEnvValue(apiKeyNames),
  };
}

export function buildAuthorizationHeaders(
  credentials: ApiCredentials | null | undefined,
): Record<string, string> {
  const normalized = normalizeCredentials(credentials ?? null);
  if (normalized.authorization) {
    return { Authorization: normalized.authorization };
  }
  if (normalized.apiKey) {
    if (/^Bearer\s+/i.test(normalized.apiKey)) {
      return { Authorization: normalized.apiKey };
    }
    return { "X-API-Key": normalized.apiKey };
  }
  return {};
}

export function resolveAuthHeaders(
  preferred?: ApiCredentials | null,
): Record<string, string> {
  const attempts: Array<ApiCredentials | null> = [];
  if (preferred) {
    attempts.push(normalizeCredentials(preferred));
  }
  const cached = getCachedCredentials();
  if (cached) {
    attempts.push(cached);
  }
  attempts.push(getEnvCredentials());

  for (const candidate of attempts) {
    if (!candidate) {
      continue;
    }
    const headers = buildAuthorizationHeaders(candidate);
    if (Object.keys(headers).length > 0) {
      return headers;
    }
  }
  return {};
}

export { STORAGE_KEY };
