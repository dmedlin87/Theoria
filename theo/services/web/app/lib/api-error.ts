export type ApiErrorDetail = {
  message: string;
  code?: string | null;
  metadata?: Record<string, unknown> | null;
  traceId?: string | null;
};

export class ApiError extends Error {
  readonly status: number;
  readonly code?: string;
  readonly metadata?: Record<string, unknown> | null;
  readonly traceId?: string | null;

  constructor(
    status: number,
    detail: ApiErrorDetail,
    options?: { cause?: unknown },
  ) {
    super(detail.message, options);
    this.name = "ApiError";
    this.status = status;
    this.code = detail.code ?? undefined;
    this.metadata = detail.metadata ?? null;
    this.traceId = detail.traceId ?? null;
  }
}

export function isApiError(error: unknown): error is ApiError {
  return error instanceof ApiError;
}
