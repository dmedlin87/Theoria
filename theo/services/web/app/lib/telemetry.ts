"use client";

type TelemetryEventInput = {
  event: string;
  durationMs: number;
  workflow?: string;
  metadata?: Record<string, unknown>;
};

type TelemetryPayload = {
  events: Array<{
    event: string;
    duration_ms: number;
    workflow?: string;
    metadata?: Record<string, unknown>;
  }>;
  page?: string | null;
};

export async function emitTelemetry(
  events: TelemetryEventInput[],
  context?: { page?: string },
): Promise<void> {
  if (events.length === 0) {
    return;
  }
  const payload: TelemetryPayload = {
    events: events.map((event) => ({
      event: event.event,
      duration_ms: Math.max(0, event.durationMs),
      workflow: event.workflow,
      metadata: event.metadata,
    })),
    page: context?.page ?? undefined,
  };
  try {
    await fetch("/api/analytics/telemetry", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload),
    });
  } catch (error) {
    if (process.env.NODE_ENV !== "production") {
      console.debug("Failed to emit telemetry", error);
    }
  }
}
