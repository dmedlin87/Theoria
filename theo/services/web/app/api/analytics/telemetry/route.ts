import { NextRequest, NextResponse } from "next/server";

import { getApiBaseUrl } from "../../../lib/api";
import { forwardTraceHeaders } from "../../trace";
import { createProxyErrorResponse } from "../../utils/proxyError";

export async function POST(request: NextRequest): Promise<NextResponse> {
  const baseUrl = getApiBaseUrl().replace(/\/$/, "");
  const target = new URL("/analytics/telemetry", baseUrl);
  const body = await request.text();
  try {
    const response = await fetch(target, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
      body,
      cache: "no-store",
    });
    const payload = await response.text();
    const headers = new Headers();
    headers.set("content-type", response.headers.get("content-type") ?? "application/json");
    forwardTraceHeaders(response.headers, headers);
    return new NextResponse(payload, { status: response.status, headers });
  } catch (error) {
    return createProxyErrorResponse({
      error,
      logContext: "Failed to proxy analytics telemetry request",
      message: "Telemetry service is currently unavailable. Please try again later.",
    });
  }
}
