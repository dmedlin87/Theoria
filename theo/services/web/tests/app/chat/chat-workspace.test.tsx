/** @jest-environment jsdom */

import "@testing-library/jest-dom";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import type {
  ChatWorkflowClient,
  ChatWorkflowResult,
  ChatWorkflowStreamEvent,
} from "../../../app/lib/chat-client";
import ChatWorkspace from "../../../app/chat/ChatWorkspace";
import { emitTelemetry, submitFeedback } from "../../../app/lib/telemetry";

jest.mock("../../../app/lib/telemetry", () => ({
  submitFeedback: jest.fn(),
  emitTelemetry: jest.fn(),
}));

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
    replace: jest.fn(),
  }),
}));

const sampleAnswer = {
  summary: "The prologue echoes Genesis, presenting the Logos as divine and eternal.",
  citations: [
    {
      index: 0,
      osis: "John.1.1",
      anchor: "John 1:1",
      passage_id: "passage-1",
      document_id: "doc-1",
      document_title: "Sample Commentary",
      snippet: "In the beginning was the Word, and the Word was with God.",
      source_url: null,
    },
  ],
  model_name: null,
  model_output: null,
  guardrail_profile: null,
} satisfies import("../../../app/copilot/components/types").RAGAnswer;

jest.mock("../../../app/mode-context", () => {
  const { RESEARCH_MODES, DEFAULT_MODE_ID } = require("../../../app/mode-config");
  return {
    useMode: () => ({
      mode: RESEARCH_MODES[DEFAULT_MODE_ID],
      modes: Object.values(RESEARCH_MODES),
      setMode: jest.fn(),
    }),
  };
});

describe("ChatWorkspace", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("prefills the textarea when a sample question chip is clicked", () => {
    const client: ChatWorkflowClient = {
      runChatWorkflow: jest.fn(),
      fetchChatSession: jest.fn(async () => null),
    };
    const telemetryMock = emitTelemetry as jest.MockedFunction<typeof emitTelemetry>;
    telemetryMock.mockResolvedValue(undefined);

    render(<ChatWorkspace client={client} />);

    const textarea = screen.getByLabelText("Ask Theo Engine") as HTMLTextAreaElement;
    expect(textarea).toHaveValue("");

    const sampleChip = screen.getByRole("button", {
      name: "How does John 1:1 connect with Genesis 1?",
    });

    fireEvent.click(sampleChip);

    expect(textarea).toHaveValue("How does John 1:1 connect with Genesis 1?");
    expect(telemetryMock).toHaveBeenCalledWith(
      [
        expect.objectContaining({
          event: "chat.sample_question_click",
          metadata: expect.objectContaining({ index: 0 }),
        }),
      ],
      { page: "chat" },
    );
  });

  it("streams a chat response and renders citations", async () => {
    const events: ChatWorkflowStreamEvent[] = [
      { type: "answer_fragment", content: "In the beginning" },
      { type: "complete", response: { sessionId: "session-1", answer: sampleAnswer } },
    ];
    const successResult: ChatWorkflowResult = {
      kind: "success",
      sessionId: "session-1",
      answer: sampleAnswer,
    };
    const client: ChatWorkflowClient = {
      runChatWorkflow: jest.fn(async (_payload, options) => {
        events.forEach((event) => options?.onEvent?.(event));
        return successResult;
      }),
      fetchChatSession: jest.fn(async () => null),
    };

    render(<ChatWorkspace client={client} />);

    fireEvent.change(screen.getByLabelText("Ask Theo Engine"), {
      target: { value: "Explain John 1:1" },
    });
    fireEvent.click(screen.getByRole("button", { name: "Send" }));

    await waitFor(() => {
      expect(screen.getByText(sampleAnswer.summary)).toBeInTheDocument();
    });
    expect(
      screen.getByRole("link", { name: "Open John 1:1" })
    ).toHaveAttribute("href", "/verse/John.1.1");
    expect(screen.getByRole("link", { name: "Search references" })).toHaveAttribute(
      "href",
      "/search?osis=John.1.1",
    );
    expect(client.runChatWorkflow).toHaveBeenCalledWith(
      expect.objectContaining({ modeId: "balanced" }),
      expect.objectContaining({ onEvent: expect.any(Function) }),
    );
  });

  it("surface guardrail violations", async () => {
    const guardrailResult: ChatWorkflowResult = {
      kind: "guardrail",
      message: "Blocked by safeguards",
      traceId: "trace-123",
      suggestions: [],
      metadata: null,
    };
    const client: ChatWorkflowClient = {
      runChatWorkflow: jest.fn(async (_payload, options) => {
        const guardrailEvent: ChatWorkflowStreamEvent = {
          type: "guardrail_violation",
          message: "Blocked by safeguards",
          traceId: "trace-123",
          suggestions: [],
          metadata: null,
        };
        options?.onEvent?.(guardrailEvent);
        return guardrailResult;
      }),
      fetchChatSession: jest.fn(async () => null),
    };

    render(<ChatWorkspace client={client} />);

    fireEvent.change(screen.getByLabelText("Ask Theo Engine"), {
      target: { value: "Share restricted content" },
    });
    const sendButton = screen.getByRole("button", { name: "Send" });
    await waitFor(() => expect(sendButton).not.toBeDisabled());
    fireEvent.click(sendButton);

    expect(await screen.findByText("Blocked by safeguards")).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Rephrase question" })).toBeInTheDocument();
  });

  it("sends feedback when reaction buttons are pressed", async () => {
    const events: ChatWorkflowStreamEvent[] = [
      { type: "answer_fragment", content: "In the beginning" },
      { type: "complete", response: { sessionId: "session-1", answer: sampleAnswer } },
    ];
    const successResult: ChatWorkflowResult = {
      kind: "success",
      sessionId: "session-1",
      answer: sampleAnswer,
    };
    const client: ChatWorkflowClient = {
      runChatWorkflow: jest.fn(async (_payload, options) => {
        events.forEach((event) => options?.onEvent?.(event));
        return successResult;
      }),
      fetchChatSession: jest.fn(async () => null),
    };

    const submitMock = submitFeedback as jest.MockedFunction<typeof submitFeedback>;
    submitMock.mockResolvedValue(undefined);

    render(<ChatWorkspace client={client} />);

    fireEvent.change(screen.getByLabelText("Ask Theo Engine"), {
      target: { value: "Explain John 1:1" },
    });
    fireEvent.click(screen.getByRole("button", { name: "Send" }));

    await waitFor(() => {
      expect(screen.getByText(sampleAnswer.summary)).toBeInTheDocument();
    });

    const helpfulButton = screen.getByRole("button", { name: "Mark response helpful" });
    const unhelpfulButton = screen.getByRole("button", { name: "Mark response unhelpful" });

    let resolveFeedback: (() => void) | undefined;
    const pendingFeedback = new Promise<void>((resolve) => {
      resolveFeedback = resolve;
    });
    submitMock.mockReturnValueOnce(pendingFeedback);

    fireEvent.click(helpfulButton);

    expect(helpfulButton).toBeDisabled();
    expect(unhelpfulButton).toBeDisabled();
    expect(submitMock).toHaveBeenCalledWith(
      expect.objectContaining({
        action: "like",
        chatSessionId: "session-1",
        query: "Explain John 1:1",
      }),
    );

    resolveFeedback?.();
    await waitFor(() => expect(helpfulButton).not.toBeDisabled());
    expect(helpfulButton).toHaveAttribute("aria-pressed", "true");
    expect(unhelpfulButton).toHaveAttribute("aria-pressed", "false");
  });

  it("auto-submits new prompts when initialPrompt changes", async () => {
    const client: ChatWorkflowClient = {
      runChatWorkflow: jest.fn(async () => ({
        kind: "success",
        sessionId: "session-1",
        answer: sampleAnswer,
      })),
      fetchChatSession: jest.fn(async () => null),
    };

    const { rerender } = render(
      <ChatWorkspace client={client} initialPrompt="Explain John 1" autoSubmit />,
    );

    await waitFor(() => {
      expect(client.runChatWorkflow).toHaveBeenNthCalledWith(
        1,
        expect.objectContaining({ prompt: "Explain John 1" }),
        expect.objectContaining({ onEvent: expect.any(Function) }),
      );
    });

    rerender(
      <ChatWorkspace client={client} initialPrompt="Summarize Romans 8" autoSubmit />,
    );

    await waitFor(() => {
      expect(client.runChatWorkflow).toHaveBeenNthCalledWith(
        2,
        expect.objectContaining({ prompt: "Summarize Romans 8" }),
        expect.objectContaining({ onEvent: expect.any(Function) }),
      );
    });

    expect(client.runChatWorkflow).toHaveBeenCalledTimes(2);
  });
});
