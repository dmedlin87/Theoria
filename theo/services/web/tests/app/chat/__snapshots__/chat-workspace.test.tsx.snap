// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChatWorkspace renders empty-state CTAs for new sessions 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="workspace"
  >
    <section
      aria-label="Chat overview"
      class="hero"
    >
      <div
        class="heroContent"
      >
        <p
          class="heroEyebrow"
        >
          Theoria Copilot
        </p>
        <h2>
          Ask with Balanced stance
        </h2>
        <p>
          Weâ€™ll keep the conversation aligned to 
          <strong>
            balanced
          </strong>
           emphasis while grounding every answer with citations you can inspect. Follow threads, branch ideas, and stay rooted in scripture.
        </p>
        <div
          aria-label="Quick navigation"
          class="heroActions"
        >
          <a
            class="heroAction"
            href="/search"
          >
            <span
              class="heroActionIcon"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-search icon icon--lg"
                fill="none"
                focusable="false"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21 21-4.34-4.34"
                />
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
              </svg>
            </span>
            <span
              class="heroActionCopy"
            >
              <strong>
                Search the library
              </strong>
              <span>
                Jump into cross-references
              </span>
            </span>
            <span
              class="heroActionArrow"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-arrow-right icon icon--md"
                fill="none"
                focusable="false"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="m12 5 7 7-7 7"
                />
              </svg>
            </span>
          </a>
          <a
            class="heroAction"
            href="/verse"
          >
            <span
              class="heroActionIcon"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-book-open icon icon--lg"
                fill="none"
                focusable="false"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 7v14"
                />
                <path
                  d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
                />
              </svg>
            </span>
            <span
              class="heroActionCopy"
            >
              <strong>
                Trace a passage
              </strong>
              <span>
                Explore every verse connection
              </span>
            </span>
            <span
              class="heroActionArrow"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-arrow-right icon icon--md"
                fill="none"
                focusable="false"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="m12 5 7 7-7 7"
                />
              </svg>
            </span>
          </a>
          <a
            class="heroAction"
            href="/upload"
          >
            <span
              class="heroActionIcon"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-upload icon icon--lg"
                fill="none"
                focusable="false"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3v12"
                />
                <path
                  d="m17 8-5-5-5 5"
                />
                <path
                  d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
                />
              </svg>
            </span>
            <span
              class="heroActionCopy"
            >
              <strong>
                Enrich your corpus
              </strong>
              <span>
                Upload documents for future chats
              </span>
            </span>
            <span
              class="heroActionArrow"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-arrow-right icon icon--md"
                fill="none"
                focusable="false"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="m12 5 7 7-7 7"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
      <ul
        aria-label="What this workspace offers"
        class="heroHighlights"
      >
        <li
          class="heroHighlight"
        >
          <span
            class="heroHighlightIcon"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-circle-check-big icon icon--lg"
              fill="none"
              focusable="false"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.801 10A10 10 0 1 1 17 3.335"
              />
              <path
                d="m9 11 3 3L22 4"
              />
            </svg>
          </span>
          <div>
            <p
              class="heroHighlightTitle"
            >
              Grounded answers
            </p>
            <p
              class="heroHighlightText"
            >
              Every response links back to trusted sources.
            </p>
          </div>
        </li>
        <li
          class="heroHighlight"
        >
          <span
            class="heroHighlightIcon"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-globe icon icon--lg"
              fill="none"
              focusable="false"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
              />
              <path
                d="M2 12h20"
              />
            </svg>
          </span>
          <div>
            <p
              class="heroHighlightTitle"
            >
              Perspective aware
            </p>
            <p
              class="heroHighlightText"
            >
              Tune the stance to match your research context.
            </p>
          </div>
        </li>
        <li
          class="heroHighlight"
        >
          <span
            class="heroHighlightIcon"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-file-text icon icon--lg"
              fill="none"
              focusable="false"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M10 9H8"
              />
              <path
                d="M16 13H8"
              />
              <path
                d="M16 17H8"
              />
            </svg>
          </span>
          <div>
            <p
              class="heroHighlightTitle"
            >
              Export ready
            </p>
            <p
              class="heroHighlightText"
            >
              Capture threads and build shareable briefs effortlessly.
            </p>
          </div>
        </li>
      </ul>
    </section>
    <div
      aria-label="Chat transcript"
      class="transcript"
      role="log"
    >
      <div
        class="emptyState"
      >
        <h3>
          Start the conversation
        </h3>
        <p>
          Ask about a passage, doctrine, or theme and weâ€™ll respond with cited insights.
        </p>
        <ul
          class="emptyStateActions"
        >
          <li>
            <button
              class="emptyStateChip"
              type="button"
            >
              How does John 1:1 connect with Genesis 1?
            </button>
          </li>
          <li>
            <button
              class="emptyStateChip"
              type="button"
            >
              What does Romans 8 teach about life in the Spirit?
            </button>
          </li>
          <li>
            <button
              class="emptyStateChip"
              type="button"
            >
              Summarize the Beatitudes in Matthew 5.
            </button>
          </li>
          <li>
            <button
              class="emptyStateChip"
              type="button"
            >
              Where else does Scripture describe the New Covenant?
            </button>
          </li>
        </ul>
        <div
          class="emptyStateCtas"
        >
          <button
            class="emptyStateCta"
            type="button"
          >
            Upload your first document
          </button>
          <button
            class="emptyStateCtaSecondary"
            type="button"
          >
            Open search examples
          </button>
          <button
            class="emptyStateCtaTertiary"
            type="button"
          >
            Watch intro video
          </button>
        </div>
        <p
          class="emptyStateLinks"
        >
          Prefer browsing? Explore the 
          <a
            href="/search"
          >
            Search
          </a>
           and 
          <a
            href="/verse"
          >
            Verse explorer
          </a>
          .
        </p>
      </div>
    </div>
    <div
      aria-label="Session history controls"
      class="container"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="trigger"
        data-state="closed"
        disabled=""
        id="radix-_r_0_"
        type="button"
      >
        Session actions
      </button>
    </div>
    <div
      aria-hidden="true"
    />
    <form
      aria-label="Chat input"
      class="form"
    >
      <label
        for="chat-question"
      >
        Ask Theoria
      </label>
      <textarea
        aria-describedby="chat-shortcuts"
        id="chat-question"
        name="question"
        placeholder="How does John 1:1 connect with Genesis 1?"
        required=""
        rows="4"
      />
      <p
        class="shortcuts"
        id="chat-shortcuts"
      >
        Press 
        <kbd>
          Ctrl+Enter
        </kbd>
         (or 
        <kbd>
          âŒ˜+Enter
        </kbd>
        ) to send
      </p>
      <div
        class="formActions"
      >
        <button
          disabled=""
          type="submit"
        >
          Send
        </button>
      </div>
    </form>
    <p
      class="footnote"
    >
      Responses cite the passages and sources that shaped the answer.
    </p>
  </div>
</DocumentFragment>
`;
